### ISSUE TITLE
Style/HashSyntax incorrectly output with UseHashRocketsWithSymbolValues option

### ISSUE THREAD

**Issue by chicaco** [bbatsov/rubocop#4642]
Syle/HashSyntax cop incorrectly output with `UseHashRocketsWithSymbolValues: true`.
And despite the same cord, the results are different.

--------

## .rubocop.yml

```
Style/HashSyntax:
  Enabled: true
  EnforcedStyle: ruby19_no_mixed_keys
  UseHashRocketsWithSymbolValues: true
  PreferHashRocketsForNonAlnumEndingSymbols: false
```

## Expected behavior

```
resources :photos, only: 'create'    # OK
resources :videos, :only => 'create' # NG
resources :books, only: :create      # NG
resources :movies, :only => :create  # OK

resources :comments, :shallow => true # NG
resources :messages, shallow: true    # OK

resources :users, only: 'create'     # OK
resources :groups, :only => 'create' # NG
resources :segments, only: :create   # NG
resources :units, :only => :create   # OK
```

## Actual behavior

It is reported that some OK codes are incorrect and some NG codes are correct.

```
lib/style_hash_syntax.rb:4:20: C: Style/HashSyntax: Use the new Ruby 1.9 hash syntax.
resources :videos, :only => 'create' # NG
                   ^^^^^^^^
lib/style_hash_syntax.rb:5:19: C: Style/HashSyntax: Use hash rockets syntax.
resources :books, only: :create      # NG
                  ^^^^^
lib/style_hash_syntax.rb:9:22: C: Style/HashSyntax: Use hash rockets syntax.
resources :messages, shallow: true    # OK
                     ^^^^^^^^
lib/style_hash_syntax.rb:11:19: C: Style/HashSyntax: Use hash rockets syntax.
resources :users, only: 'create'     # OK
                  ^^^^^
lib/style_hash_syntax.rb:13:22: C: Style/HashSyntax: Use hash rockets syntax.
resources :segments, only: :create   # NG
                     ^^^^^
```
## Steps to reproduce the problem

As above.

## RuboCop version

```
$ rubocop -V
0.49.1 (using Parser 2.4.0.0, running on ruby 2.4.1 x86_64-linux)
```

**Comment by Drenmi** [comment]
I believe this is due to the bug described by @jonas054 in #4229. Closing as a duplicate for now. Please follow that issue for updates. 🙂
